CMakeLists.txt

```diff
@@ -196,22 +213,28 @@ check_function_exists(pthread_rwlockattr_setkind_np HAVE_PTHREAD_RWLOCKATTR_SETK
 # dependencies - openssl
 if(ENABLE_TLS OR ENABLE_DNSSEC OR ENABLE_SSH)
     find_package(OpenSSL REQUIRED)
-    if (ENABLE_TLS)
+    if(ENABLE_TLS)
+        message(STATUS "OPENSSL found, required for TLS")
         set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DNC_ENABLED_TLS")
     endif()
+
+    #TODO target_link_libraries(netconf2 PUBLIC OpenSSL::SSL OpenSSL::Crypto)
     target_link_libraries(netconf2 ${OPENSSL_LIBRARIES})
     include_directories(${OPENSSL_INCLUDE_DIR})
 endif()
 
 # dependencies - libssh
-if(ENABLE_SSH)
+if(ENABLE_SSH AND OPENSSL_FOUND)
```

Comment:
SSH does not use OpenSSL directly to my knowledge, so if you want to have only SSH support, you do not need it. Should also probably be fixed in the `if` above.